src/plugin/any23/src/java/org/apache/nutch/any23/Any23ParseFilter.java

```diff
@@ -0,0 +1,175 @@
+import org.apache.nutch.parse.Parse;
+import org.apache.nutch.parse.ParseResult;
+import org.apache.nutch.protocol.Content;
+import org.ccil.cowan.tagsoup.XMLWriter;
+import org.ccil.cowan.tagsoup.jaxp.SAXParserImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.DocumentFragment;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+import org.xml.sax.XMLReader;
+
+/**
+ * <p>This implementation of {@link org.apache.nutch.parse.HtmlParseFilter}
+ * uses the <a href="http://any23.apache.org">Apache Any23</a> library
+ * for parsing and extracting structured data in RDF format from a
+ * variety of Web documents. The supported formats can be found at <a href="http://any23.apache.org">Apache Any23</a>.
+ * <p>In this implementation triples are written as Notation3 e.g.
+ * <code><http://www.bbc.co.uk/news/scotland/> <http://iptc.org/std/rNews/2011-10-07#datePublished> "2014/03/31 13:53:03"@en-gb .</code>
+ * and triples are identified within output triple streams by the presence of '\n'.
+ * The presence of the '\n' is a characteristic specific to N3 serialization in Any23.
+ * In order to use another/other writers implementing the
+ * <a href="http://any23.apache.org/apidocs/index.html?org/apache/any23/writer/TripleHandler.html">TripleHandler</a>
+ * interface, we will most likely need to identify an alternative data characteristic
+ * which we can use to split triples streams.</p>
+ * <p>
+ */
+public class Any23ParseFilter implements HtmlParseFilter {
+
+  /** Logging instance */
+  public static final Logger LOG = LoggerFactory.getLogger(Any23ParseFilter.class);
+
+  private Configuration conf = null;
+
+  /**
+   * Constant identifier used as a Key for writing and reading
+   * triples to and from the metadata Map field.
+   */
+  public final static String ANY23_TRIPLES = "Any23-Triples";
+
+  public static final String ANY_23_EXTRACTORS_CONF = "any23.extractors";
+
+  private static class Any23Parser {
+
+    Set<String> triples = null;
+
+    Any23Parser(String url, String htmlContent, String... extractorNames) throws TripleHandlerException {
+      triples = new TreeSet<String>();
+      try {
+        parse(url, htmlContent, extractorNames);
+      } catch (URISyntaxException e) {
+        throw new RuntimeException(e.getReason());
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+
+    /**
+     * Maintains a {@link java.util.Set} containing the triples
+     * @return a {@link java.util.Set} of triples.
+     */
+    private Set<String> getTriples() {
+      return triples;
+    }
+
+    private void parse(String url, String htmlContent, String... extractorNames) throws URISyntaxException, IOException, TripleHandlerException {
+      Any23 any23 = new Any23(extractorNames);
+      any23.setMIMETypeDetector(null);
+
+      try {
+        // Fix input to avoid extraction error (https://github.com/semarglproject/semargl/issues/37#issuecomment-69381281)
```

Comment:
Ideally this code should be ported over to Any23's BaseRDFParser implementations... i think the RDFa11 parser is an example.

Reply:
I'm going to ignore this within the scope of this merge request as we already invested many hours into it.

Reply:
I will happily do it over at Any23 in the future.